### PR TITLE
Add routing validation

### PR DESCRIPTION
**Pull request by fabxc**
The root route in the configuration must not exclude any alerts,
i.e. not have any matchers.
All receivers referenced in the routing tree are validated to exist.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -207,9 +207,35 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 		names[rcv.Name] = struct{}{}
 	}
+
+	// The root route must not have any matchers as it is the fallback node
+	// for all alerts.
+	if len(c.Route.Match) > 0 || len(c.Route.MatchRE) > 0 {
```
Comment (brian-brazil) [inline config/config.go]: c.Route could be null, we should check for that first or change it not to be a pointer

**Comment by fabxc** [comment]
Fixed

File: config/config.go [hunk 2]
```diff
@@ -207,9 +207,38 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 		names[rcv.Name] = struct{}{}
 	}
+
+	// The root route must not have any matchers as it is the fallback node
+	// for all alerts.
+    if c.Route == nil {
```
Comment (brian-brazil) [inline config/config.go]: You need a go fmt

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**